model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -155,12 +156,13 @@ def get_configs(self) -> Generator[RunConfig, None, None]:
         if logging.DEBUG:
             self._print_debug_search_space_info()
 
+        min_configs_to_search = self._determine_minimum_number_of_configs_to_search()
         max_configs_to_search = self._determine_maximum_number_of_configs_to_search()
         # TODO: TMA-1885: Need an early exit strategy
-        for trial_count in range(max_configs_to_search):
+        for trial_number in range(max_configs_to_search):
             trial = self._study.ask()
             trial_objectives = self._create_trial_objectives(trial)
-            logger.debug(f"Trial {trial_count+1} of {max_configs_to_search}:")
+            logger.debug(f"Trial {trial_number+1} of {max_configs_to_search}:")
```

Comment:
I'm noticing you are printing trial+1, but best_trial_number doesn't have the +1. Not sure it will burn us anywhere, but wanted to draw your attention to it. Wondering if it would be easier to change the range from 1:N instead of 0:N-1

Reply:
Good suggestion. I've made the change.